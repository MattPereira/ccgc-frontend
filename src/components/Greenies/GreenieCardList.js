import React from "react";
import GreenieCard from "./GreenieCard";
import { Row, Col } from "react-bootstrap";

/** Show list of greenie cards.
 *
 * Used by both GreenieList and TournamentDetails to list GreenieCards.
 *
 *
 * GreenieList -> GreenieCardList -> GreenieCard
 * TournamentDetails -> GreenieCardList -> GreenieCard
 */

const GreenieCardList = ({ greenies }) => {
  console.debug("GreenieCardList", "greenies=", greenies);

  return (
    <Row className="justify-content-center">
      {greenies.map((g) => (
        <Col xs={12} md={10} lg={6} key={g.id}>
          <GreenieCard greenie={g} />
        </Col>
      ))}
    </Row>
  );
};

export default GreenieCardList;
