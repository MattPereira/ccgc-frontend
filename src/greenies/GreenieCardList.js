import React from "react";
import GreenieCard from "./GreenieCard";

/** Show list of greenie cards.
 *
 * Used by both GreenieList and TournamentDetails to list GreenieCards.
 *
 *
 * GreenieList -> GreenieCardList -> GreenieCard
 * TournamentDetails -> GreenieCardList -> GreenieCard
 */

const GreenieCardList = ({ greenies }) => {
  console.debug("GreenieCardList", "greenies=", greenies);

  return (
    <div className="row justify-content-center">
      {greenies.map((g) => (
        <div className="col-9 col-sm-6 col-md-4 col-lg-3" key={g.id}>
          <GreenieCard
            key={g.id}
            id={g.id}
            date={g.tournamentDate}
            owner={`${g.firstName} ${g.lastName}`}
            holeNumber={g.holeNumber}
            feet={g.feet}
            inches={g.inches}
            courseName={g.courseName}
            courseImg={g.courseImg}
          />
        </div>
      ))}
    </div>
  );
};

export default GreenieCardList;
