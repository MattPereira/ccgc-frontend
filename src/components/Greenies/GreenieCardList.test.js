import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import GreenieCardList from "./GreenieCardList";

const greenies = [
  {
    id: 1,
    roundId: 1,
    holeNumber: 1,
    feet: 1,
    inches: 1,
    courseImg: null,
    courseName: "Lone Tree Golf Course",
    firstName: "Happy",
    lastName: "Gilmore",
    tournamentDate: "2022-01-01T08:00:00.000Z",
  },
  {
    id: 2,
    roundId: 2,
    holeNumber: 1,
    feet: 2,
    inches: 2,
    courseImg: null,
    courseName: "Lone Tree Golf Course",
    firstName: "Shooter",
    lastName: "McGavin",
    tournamentDate: "2022-01-01T08:00:00.000Z",
  },
];

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <GreenieCardList greenies={greenies} />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <GreenieCardList greenies={greenies} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays multiple greenie cards", function () {
  render(
    <MemoryRouter>
      <GreenieCardList greenies={greenies} />
    </MemoryRouter>
  );

  const happyCard = screen.getByText("Happy Gilmore");
  expect(happyCard).toBeInTheDocument();

  const shooterCard = screen.getByText("Shooter McGavin");
  expect(shooterCard).toBeInTheDocument();
});
