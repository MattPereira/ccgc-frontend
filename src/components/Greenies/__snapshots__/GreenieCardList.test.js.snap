// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row justify-content-center"
  >
    <div
      class="col-9 col-sm-6 col-md-4 col-lg-3"
    >
      <div
        class="mb-4"
      >
        <a
          class="TournamentCard-link"
          href="/greenies/1"
        >
          <div
            class="GreenieCard-card card"
          >
            <div
              class="bg-dark mb-0 py-2 text-white card-title h5"
              tag="h5"
            >
              Lone Tree
            </div>
            <div
              class="GreenieCard-crop"
            >
              <img
                alt="Lone Tree Golf Course"
              />
            </div>
            <div
              class="bg-success text-white h6 py-1"
            >
              <p
                class="lead card-text"
              >
                Happy Gilmore
              </p>
            </div>
            <div
              class="card-body"
            >
              <div
                class="table-responsive"
              >
                <table
                  class="table"
                >
                  <thead>
                    <tr>
                      <th>
                        Hole
                      </th>
                      <th>
                        Distance
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        #1
                      </td>
                      <td>
                        1' 1"
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      class="col-9 col-sm-6 col-md-4 col-lg-3"
    >
      <div
        class="mb-4"
      >
        <a
          class="TournamentCard-link"
          href="/greenies/2"
        >
          <div
            class="GreenieCard-card card"
          >
            <div
              class="bg-dark mb-0 py-2 text-white card-title h5"
              tag="h5"
            >
              Lone Tree
            </div>
            <div
              class="GreenieCard-crop"
            >
              <img
                alt="Lone Tree Golf Course"
              />
            </div>
            <div
              class="bg-success text-white h6 py-1"
            >
              <p
                class="lead card-text"
              >
                Shooter McGavin
              </p>
            </div>
            <div
              class="card-body"
            >
              <div
                class="table-responsive"
              >
                <table
                  class="table"
                >
                  <thead>
                    <tr>
                      <th>
                        Hole
                      </th>
                      <th>
                        Distance
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        #1
                      </td>
                      <td>
                        2' 2"
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
